ext/pdo_firebird/firebird_driver.c

```diff
@@ -492,13 +492,16 @@ static void firebird_handle_closer(pdo_dbh_t *dbh) /* {{{ */
 	}
 
 	if (H->date_format) {
-		efree(H->date_format);
+		pefree(H->date_format, dbh->is_persistent);
+		H->date_format = NULL;
```

Comment:
Since you're freeing the entire structure anyway below, you don't need to set the pointers for these 3 fields to NULL.